hack/lib/golang.sh

```diff
@@ -0,0 +1,426 @@
+  kubeedge::check::env
+  local -a targets=()
+  local binArg
+  for binArg in "$@"; do
+    targets+=("$(kubeedge::golang::get_target_by_binary $binArg)")
+  done
+  
+  if [[ ${#targets[@]} -eq 0 ]]; then
+    targets=("${KUBEEDGE_ALL_TARGETS[@]}")
+  fi
+    
+  local -a binaries
+  while IFS="" read -r binary; do binaries+=("$binary"); done < <(kubeedge::golang::binaries_from_targets "${targets[@]}")
+
+  local ldflags
+  read -r ldflags <<< "$(kubeedge::version::ldflags)"
+
+  mkdir -p ${KUBEEDGE_OUTPUT_BINPATH}
+  for bin in ${binaries[@]}; do
+    echo "building $bin"
+    local name="${bin##*/}"
+    set -x
+    go build -o ${KUBEEDGE_OUTPUT_BINPATH}/${name} -ldflags "$ldflags" $bin
+    set +x
+  done
+
+}
+
+
+KUBEEDGE_ALL_CROSS_BINARIES=(
+edgecore
+edgesite
+)
+
+kubeedge::golang::is_cross_build_binary() {
+  local key=$1
+  for bin in "${KUBEEDGE_ALL_CROSS_BINARIES[@]}" ; do
+    if [ "${bin}" == "${key}" ]; then
+      echo ${YES} 
+      return
+    fi
+  done
+  echo ${NO} 
+}
+
+KUBEEDGE_ALL_CROSS_GOARMS=(
+8
+7
+)
+
+kubeedge::golang::is_supported_goarm() {
+  local key=$1
+  for value in ${KUBEEDGE_ALL_CROSS_GOARMS[@]} ; do
+    if [ "${value}" == "${key}" ]; then
+      echo ${YES} 
+      return
+    fi
+  done
+  echo ${NO} 
+}
+
+kubeedge::golang::cross_build_place_binaries() {
+  kubeedge::check::env
+  
+  local -a targets=()
+  local goarm=${goarm:-${KUBEEDGE_ALL_CROSS_GOARMS[0]}}
+
+  for arg in "$@"; do
+      if [[ "${arg}" == GOARM* ]]; then
+        # Assume arguments starting with a dash are flags to pass to go.
+        goarm="${arg##*GOARM}"
+      else
+        if [ "$(kubeedge::golang::is_cross_build_binary ${arg})" == "${NO}" ]; then
+          echo "${arg} does not support cross build"
+          exit 1
+        fi
+        targets+=("$(kubeedge::golang::get_target_by_binary $arg)")
+      fi
+  done
+
+  if [[ ${#targets[@]} -eq 0 ]]; then
+    for bin in ${KUBEEDGE_ALL_CROSS_BINARIES[@]}; do
+        targets+=("$(kubeedge::golang::get_target_by_binary $bin)")
+    done
+  fi
+  
+  if [ "$(kubeedge::golang::is_supported_goarm ${goarm})" == "${NO}" ]; then
+    echo "GOARM${goarm} does not support cross build"
+    exit 1 
+  fi
+
+  local -a binaries
+  while IFS="" read -r binary; do binaries+=("$binary"); done < <(kubeedge::golang::binaries_from_targets "${targets[@]}")
+
+  local ldflags
+  read -r ldflags <<< "$(kubeedge::version::ldflags)"
+
+  mkdir -p ${KUBEEDGE_OUTPUT_BINPATH}
+  for bin in ${binaries[@]}; do
+    echo "cross buildding $bin GOARM${goarm}"
+    local name="${bin##*/}"
+    if [ "${goarm}" == "8" ]; then
+      set -x
+      GOARCH=arm64 GOOS="linux" CGO_ENABLED=1 CC=aarch64-linux-gnu-gcc go build -o ${KUBEEDGE_OUTPUT_BINPATH}/${name} -ldflags "$ldflags" $bin
+      set +x
+    elif [ "${goarm}" == "7" ]; then
+      set -x
+      GOARCH=arm GOOS="linux" GOARM=6 CGO_ENABLED=1 CC=arm-linux-gnueabi-gcc go build -o ${KUBEEDGE_OUTPUT_BINPATH}/${name} -ldflags "$ldflags" $bin
```

Comment:
should be GOARM=7 here. GOARM=${goarm}

Reply:
yes ,you are right , i will push a new pr.

thanks.